### PR TITLE
Update style for CupertinoRefreshControl.

### PR DESCRIPTION
**Pull request by Hixie**
All slivers must start with Sliver (or CupertinoSliver).
All widgets must have a key argument.

Also, some minor tweaks here and there to style and docs.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @xster @matthew-carroll

**Comment by matthew-carroll** [comment]
LGTM.  Does this require a breaking change announcement?

**Review by xster — COMMENTED** [review]
The principle behind it sounds good to me. Are we willing to make a breaking change for this?

Can we also add a test to enforce this? (e.g. attempt to build every single widget. If the first top render object is a RenderSliver, enforce that this widget's name has Sliver in it too or something)

File: packages/flutter/lib/src/cupertino/refresh.dart [hunk 1]
```diff
@@ -15,13 +15,14 @@ import 'icons.dart';
 
 class _CupertinoRefreshSliver extends SingleChildRenderObjectWidget {
```
[the remark this answers is not in the document]
Comment (Hixie) [reply packages/flutter/lib/src/cupertino/refresh.dart]: I don't feel strongly about the privates, happy to defer to you on what you want these to be called. Leave a message here and I'll rename them as you desire. :-)

**Comment by Hixie** [comment]
> The principle behind it sounds good to me. Are we willing to make a breaking change for this?

Yes, the number of people this will help far outweighs the number of people it affects today. (I hope!)


> Can we also add a test to enforce this? (e.g. attempt to build every single widget. If the first top render object is a RenderSliver, enforce that this widget's name has Sliver in it too or something)

Interesting idea. I'm not sure how to do something to every single widget in a test, though... any ideas?

**Comment by Hixie** [comment]
> LGTM. Does this require a breaking change announcement?

yes, it will.

**Comment by xster** [comment]
Are we allowed to use mirror in tests?

**Comment by Hixie** [comment]
dart:mirror is going away (and isn't compiled into our version of Dart anyway).

File: packages/flutter/lib/src/cupertino/refresh.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (xster) [reply packages/flutter/lib/src/cupertino/refresh.dart]: _CupertinoSliverRefresh

File: packages/flutter/lib/src/cupertino/refresh.dart [hunk 2]
```diff
@@ -53,8 +54,7 @@ class _CupertinoRefreshSliver extends SingleChildRenderObjectWidget {
 //
 // The [layoutExtentOffsetCompensation] field keeps internal accounting to
 // prevent scroll position jumps as the [layoutExtent] is set and unset.
-class _RenderCupertinoRefreshSliver
-    extends RenderSliver
+class _RenderCupertinoRefreshSliver extends RenderSliver
```
Comment (xster) [inline packages/flutter/lib/src/cupertino/refresh.dart]: _RenderCupertinoSliverRefresh

**Review by xster — APPROVED** [review]
LGTM. Thanks.

**Comment by Hixie** [comment]
I thought of a way to test that the names are correct: put an assert in the framework that checks if the render object is a sliver and if so, that the widget is has Sliver in the name. But that doesn't work, because some widgets are agnostic and can be used with both slivers and non-slivers.

**Comment by Hixie** [comment]
fixed the names as suggested above

**Comment by Hixie** [comment]
E-mail sent. Will update changelog when landing.

**Pull request merged**